### PR TITLE
Fix the tilt build locally

### PR DESCRIPTION
**Pull request by Skarlso**
Fix the timeout on OSX

Closes https://github.com/weaveworks/weave-gitops/issues/1389

```
STEP 5/5 — Deploying
     Applying YAML to cluster
     Objects applied to cluster:
       → wego-app:deployment

     Step 1 - 11.47s (Building Dockerfile: [localhost:5001/weaveworks/wego-app-tilt_docker_build_with_restart_base])
     Step 2 - 0.00s (Pushing localhost:5001/weaveworks/wego-app-tilt_docker_build_with_restart_base:tilt-3f94c43cd0d832bb)
     Step 3 - 2.63s (Building Dockerfile: [localhost:5001/weaveworks/wego-app])
     Step 4 - 13.37s (Pushing localhost:5001/weaveworks/wego-app:tilt-629a9f829efb7ba1)
     Step 5 - 0.04s (Deploying)
     DONE IN: 27.53s 
```

Acceptance test also works:

```
 => => naming to localhost:5001/wego-app:latest                                                                                  0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
docker push localhost:5001/wego-app:latest
The push refers to repository [localhost:5001/wego-app]
4471fb05fc35: Pushed
6290a80c31ae: Pushed
8d6f5bfbc69d: Pushed
0b3d0512394d: Pushed
5b1fa8e3e100: Pushed
latest: digest: sha256:baec154623d9e3cedcfdbfd046ddf000d972394d6da7a4aaf659292c321cef81 size: 1370
IS_TEST_ENV=true IS_LOCAL_REGISTRY=true ginkgo  -v ./test/acceptance/test/...
Running Suite: Weave GitOps User Acceptance Tests
=================================================
Random Seed: 1644271030
Will run 39 of 47 specs

Weave GitOps Add Tests With Long Cluster Name
  SmokeTestLong - Verify that gitops can deploy an app with config-repo set to <url>
  /Users/skarlso/goprojects/weaveworks/weave-gitops/test/acceptance/test/add_tests.go:1765
time="2022-02-07T22:57:14+01:00" level=info msg="GITOPS Binary Path: /Users/skarlso/goprojects/weaveworks/weave-gitops/bin/gitops"
time="2022-02-07T22:57:14+01:00" level=info msg="Using git provider: github"
STEP: Given I have a brand new cluster with a long cluster name
time="2022-02-07T22:57:14+01:00" level=info msg="Creating a kind cluster "
Delete existing kind clusters


```

### CODE REVIEW

**Comment by Skarlso** [comment]
```
[K8s EVENT: ReplicaSet wego-app-58bcfd7576 (ns: wego-system)] Error creating: pods "wego-app-58bcfd7576-" is forbidden: error looking up service account wego-system/wego-app-service-account: serviceaccount "wego-app-service-account" not found
```

This though ^^ Which isn't related to this, but pretty much prevents proper working, uh.

**Comment by dhwthompson** [comment]
This doesn’t seem to have changed anything for me: I can blat and re-create the `kind` cluster, and I’m still getting the same `Client.Timeout exceeded while awaiting headers` error.

Am I missing something?

**Comment by Skarlso** [comment]
@dhwthompson

Did you delete your registry + kind cluster?

**Comment by Skarlso** [comment]
Sorry, it's super late here, so I'm gonna go to bed. But make sure you remove the docker images, and the kind-registry container.

Do a `docker system prune --all` then try again. if it still doesn't work, please paste in here all the logs and details possible so I can debug tomorrow.

**Comment by dhwthompson** [comment]
Ah—yes, I hadn’t deleted the registry container properly. That seems to have fixed the problem for me. (Although, as you said, that’s unearthed the next problem.)

File: tools/kind-with-registry.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ esac
 running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
 if [ "${running}" != 'true' ]; then
   docker run \
-    -d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
```
Comment (dhwthompson) [inline tools/kind-with-registry.sh:44]: I’m assuming, based on what you’ve said outside this PR, that this change is a necessary part of the fix. Do you know (or at least have a working theory) what was going wrong on Macs when the `127.0.0.1` wasn’t in this command?

**Comment by Skarlso** [comment]
A problem for the next day. :D glad the build works. :)

**Review by dhwthompson — APPROVED** [review]
This seems to have fixed the issue, so I’m happy to approve this. I’ve got a question in there for my own understanding, and so that our future selves can fix similar problems in future.

File: tools/kind-with-registry.sh (hunk 1 above)
Comment (Skarlso) [reply tools/kind-with-registry.sh:44]: 0.0.0.0 is a non routable address while 127.0.0.1 is the right loopback address. The difference is that on Linux docker is native. But on a mac, you are running docker in a microvm.

**Review by aclevername — CHANGES_REQUESTED** [review]
```
• Failure [149.774 seconds]
Weave GitOps Profiles API
/home/runner/work/***/***/test/acceptance/test/profiles_test.go:32
  gets deployed and is accessible via the service [It]
  /home/runner/work/***/***/test/acceptance/test/profiles_test.go:67

  Timed out after 60.000s.
  Expected
      <int>: 503
  to equal
      <int>: 200

  /home/runner/work/***/***/test/acceptance/test/profiles_test.go:77
  • Failure [149.774 seconds]
Weave GitOps Profiles API
/home/runner/work/***/***/test/acceptance/test/profiles_test.go:32
  gets deployed and is accessible via the service [It]
  /home/runner/work/***/***/test/acceptance/test/profiles_test.go:67

  Timed out after 60.000s.
  Expected
      <int>: 503
  to equal
      <int>: 200

  /home/runner/work/***/***/test/acceptance/test/profiles_test.go:77
```

think there must be a mistaken somewhere

**Comment by Skarlso** [comment]
@aclevername Use an operating system that actually works :trollface:

**Comment by aclevername** [comment]
> @aclevername Use an operating system that actually works :trollface:

this is the github workflow :facepalm:

**Comment by Skarlso** [comment]
> this is the github workflow 🤦

I knew that. I'm totally paying attention to what you're writing. I just wanted to know that YOU are paying attention

**Comment by Skarlso** [comment]
This is weird. All of the tests are passing except that one.

**Review by aclevername — APPROVED** [review]

**Pull request merged**